### PR TITLE
Change block check type for index clear cache APIs

### PR DESCRIPTION
**Pull request by kotwanikunal**
### Description
- From the findings on https://github.com/opensearch-project/OpenSearch/pull/7498/files#r1191759700, the clear fieldsdata/request/query cache paths do not seem to be affected by metadata write blocks
- Lowering the block level to `METADATA_READ` to enable cache clearing with write blocks as it will help relieve resources within the system. 

### Related Issues
- #7498

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kotwanikunal** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/15296/
> * **CommitID:** [b0c1268](https://github.com/opensearch-project/OpenSearch/commit/b0c126837995d4d1c1e69f4355a17b0bb65b57ac)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

Looks like some BwC change wasn't backported correctly 
```
REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v2.8.0#mixedClusterTest' --tests "org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints" -Dtests.seed=FDA6AED85DA67D18 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=zh-HK -Dtests.timezone=Pacific/Midway -Druntime.java=19

org.opensearch.backwards.IndexingIT > testSeqNoCheckpoints FAILED
    org.opensearch.client.ResponseException: method [GET], host [http://[::1/]:37425], URI [test/_count?preference=_only_nodes%3Av2.8.0-3], status line [HTTP/1.1 500 Internal Server Error]
```

**Comment by kotwanikunal** [comment]
> > ### Gradle Check (Jenkins) Run Completed with:
> > 
> > * **RESULT:** FAILURE ❌
> > * **URL:** https://build.ci.opensearch.org/job/gradle-check/15296/
> > * **CommitID:** [b0c1268](https://github.com/opensearch-project/OpenSearch/commit/b0c126837995d4d1c1e69f4355a17b0bb65b57ac)
> >   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
> >   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?
> 
> Looks like some BwC change wasn't backported correctly
> 
> ```
> REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v2.8.0#mixedClusterTest' --tests "org.opensearch.backwards.IndexingIT.testSeqNoCheckpoints" -Dtests.seed=FDA6AED85DA67D18 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=zh-HK -Dtests.timezone=Pacific/Midway -Druntime.java=19
> 
> org.opensearch.backwards.IndexingIT > testSeqNoCheckpoints FAILED
>     org.opensearch.client.ResponseException: method [GET], host [http://[::1/]:37425], URI [test/_count?preference=_only_nodes%3Av2.8.0-3], status line [HTTP/1.1 500 Internal Server Error]
> ```

Seems related to #7375 failing on preferences.

File: server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java [hunk 1]
```diff
@@ -129,11 +129,11 @@ protected ShardsIterator shards(ClusterState clusterState, ClearIndicesCacheRequ
...
     @Override
     protected ClusterBlockException checkGlobalBlock(ClusterState state, ClearIndicesCacheRequest request) {
-        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
+        return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_READ);
     }
 
     @Override
     protected ClusterBlockException checkRequestBlock(ClusterState state, ClearIndicesCacheRequest request, String[] concreteIndices) {
-        return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, concreteIndices);
+        return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_READ, concreteIndices);
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/admin/indices/cache/clear/TransportClearIndicesCacheAction.java:137]: Looks legit (the caches seem to be transient anyway), @nknize any concerns with that?

File: CHANGELOG.md [hunk 2]
```diff
@@ -109,6 +109,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 ### Changed
 - Enable `./gradlew build` on MacOS by disabling bcw tests ([#7303](https://github.com/opensearch-project/OpenSearch/pull/7303))
 - Moved concurrent-search from sandbox plugin to server module behind feature flag ([#7203](https://github.com/opensearch-project/OpenSearch/pull/7203))
+- Block check type from METADATA_WRITE to METADATA_READ for index clear cache APIs ([#7303](https://github.com/opensearch-project/OpenSearch/pull/7303))
```
Comment (andrross) [inline CHANGELOG.md:112]: Maybe reword this as something like "Allow access to file cache clear APIs for read only indexes" ?

**Review by andrross — APPROVED** [review]

**Comment by kotwanikunal** [comment]
None of the classes that extend `TransportBroadcastByNodeAction` have unit tests added and will need a lot of mocking to make it useful. 
Integration tests cover the scenarios under this class, which are already added and up to date. 
`codecov/patch` does not recognize these tests though.

**Comment by andrross** [comment]
> None of the classes that extend `TransportBroadcastByNodeAction` have unit tests added and will need a lot of mocking to make it useful. Integration tests cover the scenarios under this class, which are already added and up to date. `codecov/patch` does not recognize these tests though.

Could you add unit tests for the blocking behavior with something like the following?

```
public class TransportClearIndicesCacheActionTest {
    private final TransportClearIndicesCacheAction action = new TransportClearIndicesCacheAction(
        mock(ClusterService.class),
        mock(TransportService.class),
        mock(IndicesService.class),
        mock(ActionFilters.class),
        mock(IndexNameExpressionResolver.class)
    );
    @Test
    public void test() {
        final ClusterBlocks.Builder builder = ClusterBlocks.builder();
        builder.addGlobalBlock(new ClusterBlock(1, "uuid", "", true, true, true, RestStatus.OK, EnumSet.of(ClusterBlockLevel.METADATA_WRITE)));
        ClusterState state = ClusterState.builder(ClusterState.EMPTY_STATE).blocks(builder).build();
        assertNull(action.checkGlobalBlock(state, new ClearIndicesCacheRequest()));
    }
}
```

**Comment by kotwanikunal** [comment]
> > None of the classes that extend `TransportBroadcastByNodeAction` have unit tests added and will need a lot of mocking to make it useful. Integration tests cover the scenarios under this class, which are already added and up to date. `codecov/patch` does not recognize these tests though.
> 
> Could you add unit tests for the blocking behavior with something like the following?
> 
> ```
> public class TransportClearIndicesCacheActionTest {
>     private final TransportClearIndicesCacheAction action = new TransportClearIndicesCacheAction(
>         mock(ClusterService.class),
>         mock(TransportService.class),
>         mock(IndicesService.class),
>         mock(ActionFilters.class),
>         mock(IndexNameExpressionResolver.class)
>     );
>     @Test
>     public void test() {
>         final ClusterBlocks.Builder builder = ClusterBlocks.builder();
>         builder.addGlobalBlock(new ClusterBlock(1, "uuid", "", true, true, true, RestStatus.OK, EnumSet.of(ClusterBlockLevel.METADATA_WRITE)));
>         ClusterState state = ClusterState.builder(ClusterState.EMPTY_STATE).blocks(builder).build();
>         assertNull(action.checkGlobalBlock(state, new ClearIndicesCacheRequest()));
>     }
> }
> ```

I was contemplating adding in everything mocked but testing just those two functions makes sense with the mock objects. 
Added tests in for both of the block check functions.
